snomed/com.b2international.snowowl.snomed.reasoner/src/com/b2international/snowowl/snomed/reasoner/request/SaveJobRequest.java

```diff
@@ -200,26 +215,42 @@ private Boolean persistChanges(final BranchContext context, final IProgressMonit
 		classificationTracker.classificationSaving(classificationId);
 
 		final SubMonitor subMonitor = SubMonitor.convert(monitor, "Persisting changes", 6);
-		final BulkRequestBuilder<TransactionContext> bulkRequestBuilder = BulkRequest.create();
+		final List<RequestBuilder<TransactionContext, ?>> bulkRequestBuilder = Lists.newArrayList();
 
 		applyChanges(subMonitor, context, bulkRequestBuilder);
-		final Request<BranchContext, CommitResult> commitRequest = SnomedRequests.prepareCommit()
-			.setBody(bulkRequestBuilder)
-			.setCommitComment(commitComment)
-			.setParentContextDescription(DatastoreLockContextDescriptions.SAVE_CLASSIFICATION_RESULTS)
-			.setAuthor(userId)
-			.build();
-
-		final CommitResult commitResult = new IdRequest<>(commitRequest)  
-			.execute(context);
+		
+		long resultTimeStamp = -1L;
+		for (List<RequestBuilder<TransactionContext, ?>> partition : Iterables.partition(bulkRequestBuilder, getCommitLimit(context))) {
+			final BulkRequestBuilder<TransactionContext> bulkRequest = BulkRequest.create();
+			partition.forEach(request -> bulkRequest.add(request));
+			
+			final Request<BranchContext, CommitResult> commitRequest = SnomedRequests.prepareCommit()
+					.setBody(bulkRequest.build())
+					.setCommitComment(commitComment)
+					.setParentContextDescription(DatastoreLockContextDescriptions.SAVE_CLASSIFICATION_RESULTS)
+					.setAuthor(userId)
+					.build();
+			
+			final CommitResult commitResult = new IdRequest<>(commitRequest).execute(context);
+			resultTimeStamp = commitResult.getCommitTimestamp();
+		}
 
-		classificationTracker.classificationSaved(classificationId, commitResult.getCommitTimestamp());
-		return Boolean.TRUE;
+		if (EffectiveTimes.UNSET_EFFECTIVE_TIME == resultTimeStamp) {
```

Comment:
Why `EffectiveTimes.UNSET_EFFECTIVE_TIME`? That constant is being used for other purposes. Please use `Commit.NO_COMMIT_TIMESTAMP` instead.